openapi/composer/openapi.yml

```diff
@@ -0,0 +1,230 @@
+        distribution:
+          type: string
+          example: 'rhel-8.3'
+        image_requests:
+          type: array
+          items:
+            $ref: '#/components/schemas/ImageRequest'
+        customizations:
+          $ref: '#/components/schemas/Customizations'
+    ImageRequest:
+      required:
+        - architecture
+        - image_type
+        - repositories
+        - upload_requests
+      properties:
+        architecture:
+          type: string
+          example: 'x86_64'
+        image_type:
+          type: string
+          example: 'ami'
+        repositories:
+          type: array
+          items:
+            $ref: '#/components/schemas/Repository'
+        upload_requests:
+          type: array
+          items:
+            $ref: '#/components/schemas/UploadRequest'
+    Repository:
+      type: object
+      required:
+        - baseurl
+      properties:
+        baseurl:
+          type: string
+          format: url
+          example: 'https://cdn.redhat.com/content/dist/rhel8/8/x86_64/baseos/os/'
+    UploadRequest:
+      oneOf:
+        - $ref: '#/components/schemas/AWSUploadRequest'
+    AWSUploadRequest:
+      type: object
+      required:
+        - type
+        - options
+      properties:
+        type:
+          type: string
+          enum: ['aws']
+        options:
+          type: object
+          required:
+            - region
+            - s3
+            - ec2
+          properties:
+            region:
+              type: string
+              example: 'eu-west-1'
+            s3:
+              type: object
+              required:
+                - access_key_id
+                - secret_access_key
+                - bucket
+              properties:
+                access_key_id:
+                  type: string
+                  example: 'AKIAIOSFODNN7EXAMPLE'
+                secret_access_key:
+                  type: string
+                  format: password
+                  example: 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+                bucket:
+                  type: string
+                  example: 'my-bucket'
+            ec2:
+              type: object
+              required:
+                - access_key_id
+                - secret_access_key
+              properties:
+                access_key_id:
+                  type: string
+                  example: 'AKIAIOSFODNN7EXAMPLE'
+                secret_access_key:
+                  type: string
+                  format: password
+                  example: 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+    Customizations:
+      type: object
+      properties:
+        subscription:
+          type: object
+          required:
+            - organization
+            - activation-key
+            - server-url
+            - base-url
+            - insights
+          properties:
+            organization:
+              type: integer
+              example: 2040324
+            activation-key:
```

Comment:
Underscores are used in other parts of this spec, we should be consistent in this. Same for `server-url` and `base-url`.

Reply:
Fixed.